src/org/opensolaris/opengrok/history/FileHistoryCache.java

```diff
@@ -309,12 +303,45 @@ private void storeFile(History histNew, File file, Repository repo) throws Histo
                 }
             }
         } catch (IOException ex) {
-            // Ideally we would want to catch the case when incremental update
-            // is done but the cached file is not there however we do not have
-            // the data to do it here.
+            LOGGER.log(Level.SEVERE, "Cannot open history cache file " + cacheFile.getPath());
+        }
+
+        return history;
+    }
+
+    /**
+     * Store history object (encoded as XML and compressed with gzip) in a file.
+     *
+     * @param history history object to store
+     * @param file file to store the history object into
+     * @param repo repository for the file
+     * @param mergeHistory whether to merge the history with existing or
+     *                     store the histNew as is
+     * @throws HistoryException
+     */
+    private void storeFile(History histNew, File file, Repository repo,
+            boolean mergeHistory) throws HistoryException {
+
```

Comment:
It would be nice if there was a overloaded version without the `mergeHistory` parameter.
```java
private void storeFile(History histNew, File file, Repository repo)
```

just an idea